src/sonic-yang-models/yang-models/sonic-system-port.yang

```diff
@@ -0,0 +1,81 @@
+module sonic-system-port {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-system-port";
+    prefix system-port;
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    description "SYSTEM_PORT YANG Module for SONiC OS";
+
+    revision 2022-11-10 {
+        description "First Revision";
+    }
+
+    container sonic-system-port {
+
+        container SYSTEM_PORT {
+
+            description "SYSTEM_PORT table of config_db.json";
+
+            list SYSTEM_PORT_LIST {
+
+                key "hostname asic_name ifname";
+
+                leaf hostname {
+                    type stypes:hostname;
+                }
+
+                leaf asic_name {
+                    type string;
+                }
+
+                leaf ifname {
+                    type string;
```

Comment:
Does it need to be a leafref to PORT\Interface?

Reply:
Yes, I can use a leafref. I had it as follows previously but ran into a build problem. Let's try again.

Reply:
I encountered the same build failure; e.g.
`ERROR:YANG-TEST: Exception >Leafref "/sonic-port:sonic-port/sonic-port:PORT/sonic-port:PORT_LIST/sonic-port:name" of value "Ethernet0" points to a non-existing leaf. in not empty< in /sonic/src/sonic-yang-models/tests/yang_model_tests/test_yang_model.py:208`

Reply:
You need to add Ethernet0 in your json config. For example:
https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/tests/yang_model_tests/tests_config/lldp.json#L82-L109

Reply:
Thanks @ganglyu for the tip.